pkg/controller/workload/job/job_controller.go

```diff
@@ -173,6 +238,10 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		if jobFinished {
 			return ctrl.Result{}, nil
 		}
+		if parentWorkload(&job) != "" {
+			log.V(3).Info("The parent workload '%s' does not exist", parentWorkload(&job))
+			return ctrl.Result{}, nil
+		}
 		err := r.handleJobWithNoWorkload(ctx, &job)
 		if err != nil {
 			log.Error(err, "Handling job with no workload")
```

Comment:
3. handle a finished job

should not be done for jobs with a parent workload.

Reply:
Fixed